packages/ts-types/src/narrowing/as.ts

```diff
@@ -97,16 +97,16 @@ export function asObject<T = object>(value: unknown, defaultValue?: T): Optional
  *
  * @param value The value to test.
  */
-export function asPlainObject<T = object>(value: unknown): Optional<T>;
+export function asPlainObject<T extends object = object>(value: unknown): Optional<T>;
 /**
  * Narrows an `unknown` value to an `object` if it is type-compatible, or returns the provided default otherwise.
  *
  * @param value The value to test.
  * @param defaultValue The default to return if `value` was undefined or of the incorrect type.
  */
-export function asPlainObject<T = object>(value: unknown, defaultValue: T): T;
+export function asPlainObject<T extends object = object>(value: unknown, defaultValue: T): T;
 // underlying function
-export function asPlainObject<T = object>(value: unknown, defaultValue?: T): Optional<T> {
+export function asPlainObject<T extends object = object>(value: unknown, defaultValue?: T): Optional<T> {
```

Comment:
These changes are also bug fixes that could be considered breaking changes, but they are very obscure and unlikely to have found use...